---
title: 'Data analysis: Nup98 and its interacting DExH/D-box helicases bind similar mRNAs and gene loci'
output:
  html_notebook: default
  html_document: default
  pdf_document: default
---

```{r, message=FALSE,results='hide',warning=FALSE}
library(knitr)
library(ggplot2)
#library(reshape2)
#library(plyr)
#library(dplyr)
library(data.table)
install.packages("glue")
library(dtplyr)
library(readr)
#library(GeneOverlap)
library(coin)

#Make sure your working directory in R is the folder containing the data and this files
setwd("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview")
```

Looking at commonalities between data available for Nup98 (ChIP-seq, RIP-seq, transcriptome) and other DExH/D-box protein (DBPs) that were identified in the Nup98 IP MS as Nup98 interactors. More information available in the reference below:

Capitanio JS, Montpetit B, Wozniak RW. Human Nup98 regulates the localization and activity of DExH/D-box helicase DHX9. Elife [Internet] 2017 [cited 2017 Feb 23]; 6. Available from: http://www.ncbi.nlm.nih.gov/pubmed/28221134

A summary table of DBPs interacting with Nup98 and the number of unique peptide hits detected for each helicase in the Nup98 co-IP MS is shown below.
```{r}
DBP <- c("DDX21","DDX5","DDX17","DHX15","DDX3","DHX9")
Hits <- c(26,18,15,10,10,15)
Helics <- cbind(DBP,Hits)
kable(Helics[order(-Hits),] ,format = "markdown", align = "l", padding = 2)
```
```{r}
rm(Helics)
rm(DBP)
rm(Hits)
```

I'm focusing only on data available from human cell lines or tissues. The available data is shown below.

## Data available for Nup98:

* RNA immunoprecipitation in K562 cells, [GSE67963](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE67963), [PMID:26883116](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0878-3#Sec13).

Hendrickson DG, Kelley DR, Tenen D, Bernstein B, Rinn JL. Widespread RNA binding by chromatin-associated proteins. Genome Biol [Internet] 2016 [cited 2016 Dec 18]; 17:28. Available from: http://www.ncbi.nlm.nih.gov/pubmed/26883116

* DamID-seq in HeLa cells, [GSE83692](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE83692), [PMID:27198230](http://genesdev.cshlp.org/content/30/10/1155.long).

Franks TM, Benner C, Narvaiza I, Marchetto MCN, Young JM, Malik HS, Gage FH, Hetzer MW. Evolution of a transcriptional regulator from a transmembrane nucleoporin. Genes Dev [Internet] 2016 [cited 2016 Sep 2]; 30:1-17. Available from: http://www.ncbi.nlm.nih.gov/pubmed/27198230

## Data available for DDX21:

* ChIP-seq in HEK293 cells, [GSE56802](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE56802), [PMID:25470060](https://www.ncbi.nlm.nih.gov/pubmed/25470060).
* iCLIP-Seq in HEK293 cells,[GSE56802](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE56802), [PMID:25470060](https://www.ncbi.nlm.nih.gov/pubmed/25470060).

Calo E, Flynn RA, Martin L, Spitale RC, Chang HY, Wysocka J. RNA helicase DDX21 coordinates transcription and ribosomal RNA processing. Nature [Internet] 2014 [cited 2017 Jun 19]; 518:249-53. Available from: http://www.ncbi.nlm.nih.gov/pubmed/25470060

## Data available for DDX5:

* ChIP-seq in HeLa cells, [GSE24126](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE24126), [PMID:20966046](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE24126)

Yao H, Brick K, Evrard Y, Xiao T, Camerini-Otero RD, Felsenfeld G. Mediation of CTCF transcriptional insulation by DEAD-box RNA-binding protein p68 and steroid receptor RNA activator SRA. Genes Dev [Internet] 2010 [cited 2017 Jun 19]; 24:2543-55. Available from: http://www.ncbi.nlm.nih.gov/pubmed/20966046

NOTE: No data available for DDX17 or DHX15

## Data available for DDX3(X)

* iCLIP-seq in HEK293-Flp-In-T-Rex cells, [GSE70804](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE70804), [PMID:27058758](https://www.ncbi.nlm.nih.gov/pubmed/27058758). 

Oh S, Flynn RA, Floor SN, Purzner J, Martin L, Do BT, Schubert S, Vaka D, Morrissy S, Li Y, et al. Medulloblastoma-associated DDX3 variant selectively alters the translational response to stress. Oncotarget [Internet] 2016 [cited 2017 Jun 19]; 7:28169-82. Available from: http://www.ncbi.nlm.nih.gov/pubmed/27058758

## Data available for DHX9

* RIP-seq of DHX9 in TC32 cells, 
Kindly provided by Drs. Hayriye Erkizan and Jeffrey Toretsky (Georgetown University) [PMID:25564528](https://www.ncbi.nlm.nih.gov/pubmed/25564528).

Erkizan H V, Schneider JA, Sajwan K, Graham GT, Griffin B, Chasovskikh S, Youbi SE, Kallarakal A, Chruszcz M, Padmanabhan R, et al. RNA helicase A activity is inhibited by oncogenic transcription factor EWS-FLI1. Nucleic Acids Res [Internet] 2015; 43:1069-80. Available from: http://www.ncbi.nlm.nih.gov/pubmed/25564528

## Control Datasets:

## DDX6

* eCLIP-seq in HepG2 cells, data from ENCODE. Data produced by Gene Yeo Laboratory, UCSD. Experiments: ENCSR141OIM and ENCSR099KAK (control)

## DDX59

* eCLIP-seq in HepG2 cells, data from ENCODE. Data produced by Gene Yeo Laboratory, UCSD. Experiments: ENCSR214BZA and ENCSR334OMC (control)

References for ENCODE project:

Sloan CA, Chan ET, Davidson JM, Malladi VS, Strattan JS, Hitz BC, Gabdank I, Narayanan AK, Ho M, Lee BT, et al. ENCODE data at the ENCODE portal. Nucleic Acids Res [Internet] 2016 [cited 2017 Jun 19]; 44:D726-32. Available from: http://www.ncbi.nlm.nih.gov/pubmed/26527727

Consortium EP. An integrated encyclopedia of DNA elements in the human genome. Nature [Internet] 2012; 489:57-74. Available from: http://www.ncbi.nlm.nih.gov/pubmed/22955616

## DDX11

* ChIP-seq in A549 cells, [GSE81600](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE81600), [PMID:27477908](https://www.ncbi.nlm.nih.gov/pubmed/27477908). 

Marchese F, Grossi E, Marín-Béjar O, Bharti S, Raimondi I, González J, Martínez-Herrera D, Athie A, Amadoz A, Brosh R, et al. A Long Noncoding RNA Regulates Sister Chromatid Cohesion. Mol Cell [Internet] 2016 [cited 2017 Jun 19]; 63:397-407. Available from: http://www.ncbi.nlm.nih.gov/pubmed/27477908


# RNA-immunoprecipitation of DBPs and Nup98

Let's start downloading and comparing the datasets available.

#### OF NOTE:

This document refers to the part of the analysis that was not described in the publications the original datasets came from. 
The steps before this were performed according to what was done in each paper, see the references above. For genome annotation I used Ensembl and UCSC genome browser (GRCh37/hg19).

General summary of steps done before analysis shown in document:

- Collect fastq sequencing reads
- Quality control (remove low quality reads)
- Identify barcode adapters and de-multiplex reads for each run
- Trim of 5' and 3' adapter sequences (including double ligations)
- Identify sequences that map to repetitive elements and remove
- Map unique reads to the genome
- Remove PCR duplicates and export alignment files
- Merge technical replicates and export visualization files (coverage, etc)
- Identify regions of enrichment and map them to genome annotation*

*This step has some variability depending on the experimental design used, but the basic idea is to identify regions of enrichment for protein interaction with the transcripts. Some publications used peak calling (CLIPer), others used differential abundance (Cuffdiff), others look at differences in coverage after normalization (bamCompare), etc. 

These will all give enrichment values or scores that are an indicator of the abundance of a given sequence interval in the sample after normalization to the input sample. 

This is what was used as a starting point in the analysis done in this file. 

## Load the data of RIP-seq mRNA regions annotated with ENSEMBL genes

For each dataset:

* Load dataset;
* Remove the incomplete rows (i.e. RIP regions that don't map to genes or genes without mapped reads from RIP-seq); 
* Join plus and minus strand reads or replicate reads; 
* In a separate table aggragate by gene;
* When necessary (Nup98 and DHX9) remove the interactors that show a p-value above 0.05 or fold-change under 1.5 - other samples were prefiltered before import, so this step is not necessary;
```{r, message=FALSE,results='hide',warning=FALSE}
RHA_RIP <- read_csv("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/RIPseq/RIP-DHX9/RHA_RIP_genesAnno.csv", col_types = cols(X1 = col_skip()))

# Fix the FCs, replace the 0 from block with 0.001
RHA_RIP$RHA_BLOCK_0[RHA_RIP$RHA_BLOCK_0 == 0] <- 0.001
RHA_RIP$FC <- RHA_RIP$RHA_DMSO_0 / RHA_RIP$RHA_BLOCK_0

RHA_RIP <- RHA_RIP[RHA_RIP$FC >  1.5,]

#Fix the gene and transcript names aggregate by gene, summing the RHA signal 
RHA_RIP$gene_id.y <- sapply(strsplit(as.character(RHA_RIP$gene_id.y), "\\."), "[[", 1)

DHX9v2RIPdt <- data.table(RHA_RIP)
DHX9v2RIPgeneDT <- DHX9v2RIPdt[,.(RHA.block=mean(RHA_BLOCK_0),RHA.Pos=mean(RHA_DMSO_0),RIP.FC=max(FC),count=.N),by=.(gene_id.y)]

rm(RHA_RIP)
rm(DHX9v2RIPdt)
```

Now load the data of annotated peaks from the DDX3 CLIP-seq dataset and process as described above for DHX9 until I have a table of genes.
```{r, message=FALSE,results='hide'}
DDX3wtCLIP_ensemblGenes <- read_delim("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/RIPseq/GSE70801-RIP DDX3/DDX3wtCLIP_ensemblGenes", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 1)
```

```{r}
DDX3v2RIP <- DDX3wtCLIP_ensemblGenes[complete.cases(DDX3wtCLIP_ensemblGenes),]
rm(DDX3wtCLIP_ensemblGenes)

coltitles <- names(DDX3v2RIP)
coltitles[1:4] <- c("RIP.chrom","RIP.start","RIP.end","RIP.value")
colnames(DDX3v2RIP) <- coltitles
rm(coltitles)

DDX3v2RIP$RIP.value <- sapply(strsplit(as.character(DDX3v2RIP$RIP.value), "\\,"), function(x) max(as.numeric(x)))

DDX3v2RIPdt <- data.table(DDX3v2RIP)
DDX3v2RIPgeneDT <- DDX3v2RIPdt[,.(RIP.start=min(RIP.start),RIP.end=max(RIP.end),RIP.value=max(RIP.value),count=.N),by=.(ensGene.name2,ensGene.chrom,ensGene.strand)]

rm(DDX3v2RIP)
rm(DDX3v2RIPdt)
```

Up next let's do all the same for DDX21 CLIP-seq data:
```{r, message=FALSE,results='hide'}
DDX21wtCLIP_ensemblGenes <- read_delim("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/RIPseq/GSE56802-RIP DDX21/DDX21wtCLIP_ensemblGenes", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 1)
```

```{r}
DDX21RIP <- DDX21wtCLIP_ensemblGenes[complete.cases(DDX21wtCLIP_ensemblGenes),]
rm(DDX21wtCLIP_ensemblGenes)
coltitles <- names(DDX21RIP)
coltitles[1:4] <- c("RIP.chrom","RIP.start","RIP.end","RIP.value")
colnames(DDX21RIP) <- coltitles
rm(coltitles)
DDX21RIPdt <- data.table(DDX21RIP)
DDX21RIPgeneDT <- DDX21RIPdt[,.(RIP.start=min(RIP.start),RIP.end=max(RIP.end),RIP.value=max(RIP.value),count=.N),by=.(ensGene.name2,ensGene.chrom,ensGene.strand)]

rm(DDX21RIP)
rm(DDX21RIPdt)
```

Let's do Nup98 now. For this, since it was a regular RNA-IP and not the CLIP-seq or FAST-seq the processing was done differently, so I'll just import the resulting tables here. I'm using the table by genes here.
```{r, message=FALSE,results='hide'}
N98RIP_genes <- read_csv("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/RIPseq/GSE67963-RIP N98/GSE67963_significance_calls_genes.csv")

```

```{r}
#Fix the gene and transcript names
N98RIP_genes$test_id <- sapply(strsplit(as.character(N98RIP_genes$test_id), "\\."), "[[", 1)

#Get the total number of genes from ensemble to be used in later tests. Then keep only statistically significant rows of Nup98 data
TTgenes <- dim(N98RIP_genes)[1]
N98RIP_genes.05 <- N98RIP_genes[N98RIP_genes$p_value < 0.05,]
rm(N98RIP_genes)

#Move Nup98 bound gene list to a vector
N98RIP_gen.05 <- N98RIP_genes.05$test_id
```

Let's save all this data before we do any analysis. then let's remove that while we do all the same processing for the control dataset.
```{r}
save.image(file = "RIP-seq tables Nucleus.RData")
rm(list=ls(globalenv()))
```

We have 2 control datasets, DDX6 and DDX59 CLIP-seq data from ENCODE. 

Let's start by loading all of them and removing incomplete rows
```{r, message=FALSE,results='hide'}
DDX6 <- read_delim("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/RIPseq/Cnegs ENCODE/ensemblGenes_DDX6intersect", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 1)
DDX6 <- DDX6[complete.cases(DDX6),]

DDX59 <- read_delim("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/RIPseq/Cnegs ENCODE/ensemblGenes_DDX59intersect", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 1)
DDX59 <- DDX59[complete.cases(DDX59),]
```

## Analyzing and comparing RIP-seq dataset


We'll beging by aggregating the data by gene ID:
```{r}
DDX6 <- data.table(DDX6)
DDX59 <- data.table(DDX59)

colnames(DDX6)[1] <- "ensGene.name"
colnames(DDX59)[1] <- "ensGene.name"

coltitles <- names(DDX6)
coltitles[10:15] <- c("RIP.chrom","RIP.start","RIP.end","RIP.name","RIP.score","RIP.strand")
colnames(DDX6) <- coltitles
colnames(DDX59) <- coltitles

rm(coltitles)
```

```{r}
DDX6RIP <- DDX6[,.(RIP.start=min(RIP.start),RIP.end=max(RIP.end),RIP.score=max(RIP.score),count=.N),by=.(ensGene.name2,ensGene.chrom,ensGene.strand)]
rm(DDX6)

DDX59RIP <- DDX59[,.(RIP.start=min(RIP.start),RIP.end=max(RIP.end),RIP.score=max(RIP.score),count=.N),by=.(ensGene.name2,ensGene.chrom,ensGene.strand)]
rm(DDX59)
```

Now save the control datasets as well.
```{r}
#Save this as the control dataset
save.image(file = "RIP-seq tables CONTROLS Nucleus.RData")
```

Let's bring back the data from before, of the samples from DBPs that interact with Nup98.
```{r}
load("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/RIP-seq tables Nucleus.RData")
```

## Do DBPs that bind Nup98 interact more with mRNAs that are bound by Nup98 in comparison to mRNAs that do not interact with Nup98?

Let's get some samples of the counts for genes that interact with Nup98 and those that don't from the DBP datasets. I'm using bootstrapping to collect several samples from the same dataset here. This is necessary because the number of genes whose mRNA bind to Nup98 is much smaller than the number of genes whose mRNA does not interact with Nup98. This allows us to make the datasets the same size, and the repeated sampling allows us to ensure we got representative samples of the genes whose mRNAs don't interact with Nup98.

First DDX21, using counts of intervals mapped to the same gene as our quantitative variable and separating the genes encoding the mRNAs bound by DDX21 into 2 categories: genes whose mRNAs also interact with Nup98 and those that don't.

### DDX21
```{r}
#For genes that bind Nup98
DDX21RIPgeneDT$test <- (DDX21RIPgeneDT$ensGene.name2 %in% N98RIP_gen.05)

x = DDX21RIPgeneDT$count[DDX21RIPgeneDT$test == TRUE]
n = sum(DDX21RIPgeneDT$test == TRUE)*2
B = 10000

DDX21wNup98resamples = matrix(sample(x, n * B, replace = TRUE), B, n)
DDX21wNup98resampledMeans = apply(DDX21wNup98resamples, 1, mean)
rm(DDX21wNup98resamples)

#For genes that don't bind Nup98
x = DDX21RIPgeneDT$count[DDX21RIPgeneDT$test == FALSE]
n = sum(DDX21RIPgeneDT$test == TRUE)*2
B = 10000

## bootstrap resamples
DDX21noNup98resamples = matrix(sample(x, n * B, replace = TRUE), B, n)
DDX21noNup98resampledMeans = apply(DDX21noNup98resamples, 1, mean)
rm(DDX21noNup98resamples)
```

```{r}
par(mfrow=c(1,2))
hist(DDX21wNup98resampledMeans,main = "Interaction with Nup98",xlab = "")
mtext(text="Number of DDX21 interacting \n mRNA intervals per gene", side=1, line=3)
hist(DDX21noNup98resampledMeans,main = "No Nup98 interaction",xlab = "")
mtext(text="Number of DDX21 interacting \n mRNA intervals per gene", side=1, line=3) 
```

Let's use a permutation test on those count values for our hypothesis test. We'd like to see if there is a difference in DDX21 mRNA interaction between the genes whose mRNAs also bind Nup98, versus those that do not. If these proteins (Nup98 and DDX21) function in a complex, we would expect to see more DDX21 interaction with mRNAs that also bind Nup98.
The permutation test will take a sample of the count values from all DDX21 interacting mRNAs (independently of whether or not they also bind Nup98) and randomly assign a label a these values (yes or no for also binding Nup98), then it will compare this generated data to the real data with the correct labels to see if the difference is statistically significant. This process is repeated 100000 times to reach a p-value estimate.

Note: since this is a sampling methodology the p-values obtained will not always be identical, but should be very similar.
```{r}
independence_test(DDX21RIPgeneDT$count ~ DDX21RIPgeneDT$test, distribution = approximate(B=100000))
```

Now DHX9, let's repeat all the analysis done above for DDX21 woth the DHX9 dataset.

### DHX9
```{r}
#For genes that bind Nup98
DHX9v2RIPgeneDT$test <- (DHX9v2RIPgeneDT$gene_id.y %in% N98RIP_gen.05)

x = DHX9v2RIPgeneDT$count[DHX9v2RIPgeneDT$test == TRUE]
n = sum(DHX9v2RIPgeneDT$test == TRUE)*2
B = 10000

DHX9v2wNup98resamples = matrix(sample(x, n * B, replace = TRUE), B, n)
DHX9v2wNup98resampledMeans = apply(DHX9v2wNup98resamples, 1, mean)
rm(DHX9v2wNup98resamples)
#For genes that don't bind Nup98
x = DHX9v2RIPgeneDT$count[DHX9v2RIPgeneDT$test == FALSE]
n = sum(DHX9v2RIPgeneDT$test == TRUE)
B = 10000

## bootstrap resamples
DHX9v2noNup98resamples = matrix(sample(x, n * B, replace = TRUE), B, n)
DHX9v2noNup98resampledMeans = apply(DHX9v2noNup98resamples, 1, mean)
rm(DHX9v2noNup98resamples)
```

```{r}
par(mfrow=c(1,2))
hist(DHX9v2wNup98resampledMeans,main = "Interaction with Nup98",xlab = "")
mtext(text="Number of DHX9 interacting \n mRNA intervals per gene", side=1, line=3)
hist(DHX9v2noNup98resampledMeans,main = "No Nup98 interaction",xlab = "")
mtext(text="Number of DHX9 interacting \n mRNA intervals per gene", side=1, line=3)
```

Statistical test for DHX9:
```{r}
independence_test(DHX9v2RIPgeneDT$count ~ DHX9v2RIPgeneDT$test, distribution = approximate(B=100000))
```

Repeating the procedure for DDX3.

### DDX3
```{r}
#For genes that bind Nup98
DDX3v2RIPgeneDT$test <- (DDX3v2RIPgeneDT$ensGene.name2 %in% N98RIP_gen.05)

x = DDX3v2RIPgeneDT$count[DDX3v2RIPgeneDT$test == TRUE]
n = sum(DDX3v2RIPgeneDT$test == TRUE)
B = 10000

DDX3v2wNup98resamples = matrix(sample(x, n * B, replace = TRUE), B, n)
DDX3v2wNup98resampledMeans = apply(DDX3v2wNup98resamples, 1, mean)
rm(DDX3v2wNup98resamples)
#For genes that don't bind Nup98
x = DDX3v2RIPgeneDT$count[DDX3v2RIPgeneDT$test == FALSE]
n = sum(DDX3v2RIPgeneDT$test == TRUE)
B = 10000

## bootstrap resamples
DDX3v2noNup98resamples = matrix(sample(x, n * B, replace = TRUE), B, n)
DDX3v2noNup98resampledMeans = apply(DDX3v2noNup98resamples, 1, mean)
rm(DDX3v2noNup98resamples)
```

```{r}
par(mfrow=c(1,2))
hist(DDX3v2wNup98resampledMeans,main = "Interaction with Nup98",xlab = "")
mtext(text="Number of DDX3 interacting \n mRNA intervals per gene", side=1, line=3)
hist(DDX3v2noNup98resampledMeans,main = "No Nup98 Interaction",xlab = "")
mtext(text="Number of DDX3 interacting \n mRNA intervals per gene", side=1, line=3)
```

Statistical test for DDX3:
```{r}
independence_test(DDX3v2RIPgeneDT$count ~ DDX3v2RIPgeneDT$test, distribution = approximate(B=100000))
```

### Do we observe different results if we repeat the analysis above for mRNAs bound by control DBPs, that do not interact with Nup98?

Let's repeat the analysis for 2 control dataset from DBPs that don't interact with Nup98:

### DDX6:
```{r}
#For genes that bind Nup98
DDX6RIP$test <- (DDX6RIP$ensGene.name2 %in% N98RIP_gen.05)

x = DDX6RIP$count[DDX6RIP$test == TRUE]
n = sum(DDX6RIP$test == TRUE)
B = 10000

DDX6wNup98resamples = matrix(sample(x, n * B, replace = TRUE), B, n)
DDX6wNup98resampledMeans = apply(DDX6wNup98resamples, 1, mean)
rm(DDX6wNup98resamples)
#For genes that don't bind Nup98
x = DDX6RIP$count[DDX6RIP$test == FALSE]
n = sum(DDX6RIP$test == TRUE)
B = 10000

## bootstrap resamples
DDX6noNup98resamples = matrix(sample(x, n * B, replace = TRUE), B, n)
DDX6noNup98resampledMeans = apply(DDX6noNup98resamples, 1, mean)
rm(DDX6noNup98resamples)
```

```{r}
par(mfrow=c(1,2))
hist(DDX6wNup98resampledMeans,main = "Interaction with Nup98",xlab = "")
mtext(text="Number of DDX6 interacting \n mRNA intervals per gene", side=1, line=3)
hist(DDX6noNup98resampledMeans,main = "No Nup98 interaction",xlab = "")
mtext(text="Number of DDX6 interacting \n mRNA intervals per gene", side=1, line=3)
```

Statistical test for DDX6:
```{r}
independence_test(DDX6RIP$count ~ DDX6RIP$test, distribution = approximate(B=100000))
```

### DDX59:
```{r}
#For genes that bind Nup98
DDX59RIP$test <- (DDX59RIP$ensGene.name2 %in% N98RIP_gen.05)

x = DDX59RIP$count[DDX59RIP$test == TRUE]
n = sum(DDX59RIP$test == TRUE)
B = 10000

DDX59wNup98resamples = matrix(sample(x, n * B, replace = TRUE), B, n)
DDX59wNup98resampledMeans = apply(DDX59wNup98resamples, 1, mean)
rm(DDX59wNup98resamples)
#For genes that don't bind Nup98
x = DDX59RIP$count[DDX59RIP$test == FALSE]
n = sum(DDX59RIP$test == TRUE)
B = 10000

## bootstrap resamples
DDX59noNup98resamples = matrix(sample(x, n * B, replace = TRUE), B, n)
DDX59noNup98resampledMeans = apply(DDX59noNup98resamples, 1, mean)
rm(DDX59noNup98resamples)
```

```{r}
par(mfrow=c(1,2))
hist(DDX59wNup98resampledMeans,main = "Interaction with Nup98",xlab = "")
mtext(text="Number of DDX59 interacting \n mRNA intervals per gene", side=1, line=3)
hist(DDX59noNup98resampledMeans,main = "No Nup98 interaction",xlab = "")
mtext(text="Number of DDX59 interacting \n mRNA intervals per gene", side=1, line=3)
```

```{r}
independence_test(DDX59RIP$count ~ DDX59RIP$test, distribution = approximate(B=100000))
```

## Compile and scale data for plots

We can beging by looking at the count values for mRNAs interacting with each of the DBPs. We can separate the genes into 2 groups, depending on whether or not the encoded mRNA is also bound by Nup98.

Let's try some boxplots of the data:
```{r,fig.height=5,fig.width=3.5}
boxplot(DDX21noNup98resampledMeans,DDX21wNup98resampledMeans, names = c("No", "Yes"), main = "DDX21", ylab = "", xlab = "Interaction of Nup98 with target mRNA")
mtext(text="Number of DDX21 interacting \n mRNA intervals per gene", side=2, line=2)   
```

```{r,fig.height=5,fig.width=3.5}
boxplot(DDX3v2noNup98resampledMeans,DDX3v2wNup98resampledMeans, names = c("No", "Yes"), main = "DDX3", ylab = "", xlab = "Interaction of Nup98 with target mRNA")
mtext(text="Number of DDX3 interacting \n mRNA intervals per gene", side=2, line=2)   
```

```{r,fig.height=5,fig.width=3.5}
boxplot(DHX9v2noNup98resampledMeans,DHX9v2wNup98resampledMeans, names = c("No", "Yes"), main = "DHX9", ylab = "", xlab = "Interaction of Nup98 with target mRNA")
mtext(text="Number of DHX9 interacting \n mRNA intervals per gene", side=2, line=2)   
```

Now for boxplots of the negative controls:
```{r,fig.height=5,fig.width=3.5}
boxplot(DDX6noNup98resampledMeans,DDX6wNup98resampledMeans, names = c("No", "Yes"), main = "DDX6", ylab = "", xlab = "Interaction of Nup98 with target mRNA")
mtext(text="Number of DDX6 interacting \n mRNA intervals per gene", side=2, line=2)   
```

```{r,fig.height=5,fig.width=3.5}
boxplot(DDX59noNup98resampledMeans,DDX59wNup98resampledMeans, names = c("No", "Yes"), main = "DDX59", ylab = "", xlab = "Interaction of Nup98 with target mRNA")
mtext(text="Number of DDX59 interacting \n mRNA intervals per gene", side=2, line=2)   
```

Since these are different experimental designs (RIP-seq, fRIP-seq, eCLIP-seq, FLASH-seq, HITS-seq, etc.) the sequencing data is processed differently and, despite coverage normalization, the counts value of interavals mapped to a given mRNA are not directly comparable between different DBPs.
We can get around this a little bit by centering and scaling the values from each dataset to obtain Z-scores.

First let's set the data from the bootstrapped means into tables containing the counts value and the Nup98 binding status for each DBP. Then, let's add an extra column for the scaled, centered, Z-scores for the entire dataset.

DDX21:
```{r}
Value <- DDX21noNup98resampledMeans
Nup98.bound <- rep("no",length(Value))
DBP.RIP <- rep("DDX21",length(Value))
DDX21.no <- cbind(Value,Nup98.bound,DBP.RIP)

Value <- DDX21wNup98resampledMeans
Nup98.bound <- rep("yes",length(Value))
DBP.RIP <- rep("DDX21",length(Value))
DDX21.yes <- cbind(Value,Nup98.bound,DBP.RIP)

DDX21 <- as.data.frame(rbind(DDX21.yes,DDX21.no))
DDX21$scaleT <- scale(as.numeric(as.character(DDX21[,1])),center = TRUE,scale = TRUE)
DDX21$Value <- as.numeric(as.character(DDX21$Value))
rm(DDX21.no)
rm(DDX21.yes)
```

DDX3:
```{r}
Value <- DDX3v2noNup98resampledMeans
Nup98.bound <- rep("no",length(Value))
DBP.RIP <- rep("DDX3",length(Value))
DDX3.no <- cbind(Value,Nup98.bound,DBP.RIP)

Value <- DDX3v2wNup98resampledMeans
Nup98.bound <- rep("yes",length(Value))
DBP.RIP <- rep("DDX3",length(Value))
DDX3.yes <- cbind(Value,Nup98.bound,DBP.RIP)

DDX3 <- as.data.frame(rbind(DDX3.yes,DDX3.no))
DDX3$scaleT <- scale(as.numeric(as.character(DDX3[,1])),center = TRUE,scale = TRUE)
DDX3$Value <- as.numeric(as.character(DDX3$Value))
rm(DDX3.no)
rm(DDX3.yes)
```

DHX9:
```{r}
Value <- DHX9v2noNup98resampledMeans
Nup98.bound <- rep("no",length(Value))
DBP.RIP <- rep("DHX9",length(Value))
DHX9.no <- cbind(Value,Nup98.bound,DBP.RIP)

Value <- DHX9v2wNup98resampledMeans
Nup98.bound <- rep("yes",length(Value))
DBP.RIP <- rep("DHX9",length(Value))
DHX9.yes <- cbind(Value,Nup98.bound,DBP.RIP)

DHX9 <- as.data.frame(rbind(DHX9.yes,DHX9.no))
DHX9$scaleT <- scale(as.numeric(as.character(DHX9[,1])),center = TRUE,scale = TRUE)
DHX9$Value <- as.numeric(as.character(DHX9$Value))
rm(DHX9.no)
rm(DHX9.yes)
```

DDX59:
```{r}
Value <- DDX59noNup98resampledMeans
Nup98.bound <- rep("no",length(Value))
DBP.RIP <- rep("DDX59",length(Value))
DDX59.no <- cbind(Value,Nup98.bound,DBP.RIP)

Value <- DDX59wNup98resampledMeans
Nup98.bound <- rep("yes",length(Value))
DBP.RIP <- rep("DDX59",length(Value))
DDX59.yes <- cbind(Value,Nup98.bound,DBP.RIP)

DDX59 <- as.data.frame(rbind(DDX59.yes,DDX59.no))
DDX59$scaleT <- scale(as.numeric(as.character(DDX59[,1])),center = TRUE,scale = TRUE)
DDX59$Value <- as.numeric(as.character(DDX59$Value))
rm(DDX59.no)
rm(DDX59.yes)
```

DDX6:
```{r}
Value <- DDX6noNup98resampledMeans
Nup98.bound <- rep("no",length(Value))
DBP.RIP <- rep("DDX6",length(Value))
DDX6.no <- cbind(Value,Nup98.bound,DBP.RIP)

Value <- DDX6wNup98resampledMeans
Nup98.bound <- rep("yes",length(Value))
DBP.RIP <- rep("DDX6",length(Value))
DDX6.yes <- cbind(Value,Nup98.bound,DBP.RIP)

DDX6 <- as.data.frame(rbind(DDX6.yes,DDX6.no))
DDX6$scaleT <- scale(as.numeric(as.character(DDX6[,1])),center = TRUE,scale = TRUE)

DDX6$Value <- as.numeric(as.character(DDX6$Value))
rm(DDX6.no)
rm(DDX6.yes)
```

Since the Z-scores have been calculated separately for each DBP, now we can put together a table with all the data from all the DBPs.
```{r}
All <- rbind(DDX21,DDX3,DHX9,DDX59,DDX6)

All <- rbind(DDX21,DDX3,DHX9,DDX59,DDX6)
All$scaleT <- unlist(as.numeric(as.character(All$scaleT)))
```

Let's have a look at the boxplots again to ensure the centering and scaling had no effect on the distribution of the underlying data.
```{r,fig.height=5,fig.width=7}
ggplot(All, aes(x=Nup98.bound,y=Value,fill=Nup98.bound)) + 
  geom_boxplot()+ facet_wrap(~DBP.RIP,scales = "free") +
  geom_jitter(shape=16, position=position_jitter(0.4),size=0.05,alpha=0.05)+
  xlab("Target mRNAs interaction with Nup98")+
  ylab("Number of DBP interacting mRNA intervals per gene")+
  scale_fill_hue(name="Nup98", breaks=c("yes", "no"), labels=c("yes", "no"))+
  ggtitle("Interacting Nup98 and DBPs bind similar target mRNAs")+
  theme_bw()
```

```{r,fig.height=5,fig.width=7}
ggplot(All, aes(x=Nup98.bound,y=scaleT,fill=Nup98.bound)) + 
  geom_boxplot()+ facet_wrap(~DBP.RIP,scales = "free") +
  geom_jitter(shape=16, position=position_jitter(0.4),size=0.03,alpha=0.03)+
  xlab("Target mRNAs interaction with Nup98")+
  ylab("Z-score of DBP interacting mRNA intervals per gene")+
  scale_fill_hue(name="Nup98", breaks=c("yes", "no"), labels=c("yes", "no"))+
  ggtitle("Interacting Nup98 and DBPs bind similar target mRNAs")+
  theme_bw()
```

Looks good. Remeber, the top row of boxplots contains DBPs that interact with Nup98 and the bottom row has the control DBPs that do not interact with Nup98.

Now let's clean the workspace and move on to the ChIP-seq analysis
```{r}
rm(list=ls(globalenv()))
```

# Chromatin immunoprecipitation of DBPs and Nup98

For this second part of the analysis we'll use the bed files making the identified peaks from ChIP-seq experiments described above.

```{r, message=FALSE, warning=FALSE}
#If need be:

source("http://bioconductor.org/biocLite.R")
biocLite("ChIPseeker")
biocLite("clusterProfiler")
biocLite("TxDb.Hsapiens.UCSC.hg19.knownGene")
biocLite("org.Hs.eg.db")
biocLite("GenomicFeatures")
```

```{r, message=FALSE, warning=FALSE}
library(ChIPseeker)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene
library(clusterProfiler)
library(org.Hs.eg.db)
library(GenomicFeatures)
library(regioneR)

```

Let's read the bed files containing ChIP-seq peaks for our proteins of interest:

```{r}
files <- list.files("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/ChIPseq/bed files")
files
```

```{r, message=FALSE, warning=FALSE}
setwd("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/ChIPseq/bed files")
#Make sure the names here agree with the order of files in the list above

DDX11 = readPeakFile(files[[1]])
DDX21 = readPeakFile(files[[2]])
DDX21 = DDX21[seqnames(DDX21) != 'chrM']
DDX5 = readPeakFile(files[[3]])
N98FL = readPeakFile(files[[4]])
N98Nt = readPeakFile(files[[5]])

#rm(files)
```

Next let's just have a look at the peak locations:

```{r,fig.width=10,fig.height=7, fig.height=15, fig.width=20, message=FALSE, warning=FALSE}
peaks=GenomicRanges::GRangesList(N98FL=N98FL,N98Nt=N98Nt,DDX21=DDX21,DDX5=DDX5,DDX11=DDX11)
rm(DDX11)
rm(DDX21)
rm(DDX5)
rm(N98FL)
rm(N98Nt)
rm(files)
#If you'd like to see the peaks on the chromosomes run the next line
covplot(peaks)
```

We'll create a matrix of genomic regions around transcription start sites (TSS) to see how the peaks are distributed in relation to this feature. We'll expand to 10kb in each direction from the TSS.

```{r}
promoter <- getPromoters(TxDb=txdb, upstream=10000, downstream=10000)
```

To make all plots at once:
```{r}
# Of note, you may need a computer with more than 8GB of RAM in order to run the function below.
tagMatrixList <- lapply(peaks, getTagMatrix, windows=promoter)
```
PAREI AQUI
```{r}
save(peaks,file = "all ChIP peaks as GRanges.RData")
save(tagMatrixList,file = "all ChIP peaks tagMatrix list TSS 10k.RData")
```

Let's check the distribution of the peaks in relation to the TSS of interacting genes.
```{r,fig.width=10,fig.height=10}
plotAvgProf(tagMatrixList, xlim=c(-10000, 10000),facet = "row")
```

Let's do some statistical testing to see if the overlap between Nup98 peaks and DBP peaks is significant:
```{r}
rm(tagMatrixList)
```

Let's test each Nup98 file against each DBP file:
```{r, message=TRUE, warning=TRUE}
setwd("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/ChIPseq/bed files")
files <- list.files()
# Confirm these are the correct bed files in the list
files
```
Of note, since these are sampling methods the obtained p-values will not always be the same, but they should be similar.

Full length Nup98 DamID-seq:
```{r}
setwd("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/ChIPseq/bed files")
enrichPeakOverlap(queryPeak = files[4], targetPeak = files[c(1:3,5)], TxDb = txdb, pAdjustMethod = "BH", nShuffle = 1000, pool= TRUE, chainFile = NULL, verbose = FALSE)
```

Nucleoplasmic Nup98 DamID-seq:
```{r}
setwd("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/ChIPseq/bed files")
enrichPeakOverlap(queryPeak = files[5], targetPeak = files[1:4], TxDb = txdb, pAdjustMethod = "BH", nShuffle = 1000, pool = TRUE, chainFile = NULL, verbose = FALSE)
```

Now let's annotate these peaks and look at what genomic features they interact with:
```{r}
peakAnnoList <- lapply(peaks, annotatePeak, TxDb=txdb,tssRegion=c(-3000, 3000), verbose=FALSE)
```

```{r}
plotAnnoBar(peakAnnoList)
```

Now a profile of the distance to the TSS of each peak:
```{r}
plotDistToTSS(peakAnnoList)
```

We can annotate the peaks with gene ids and do some quick enrichment analysis
```{r}
genes = lapply(peakAnnoList, function(i) as.data.frame(i)$geneId)
names(genes) = sub("_", "\n", names(genes))
```

```{r}
compKEGG <- compareCluster(geneCluster   = genes,fun= "enrichKEGG",pvalueCutoff  = 0.05,pAdjustMethod = "BH")
```

```{r,fig.width=7.5,fig.height=10}
plot(compKEGG, showCategory = 20, title = "KEGG Pathway Enrichment Analysis")
rm(compKEGG)
```

```{r}
compGOenrich <- compareCluster(geneCluster=genes, fun="enrichGO", pvalueCutoff= 0.05, OrgDb='org.Hs.eg.db')
```

```{r,fig.width=13,fig.height=15}
plot(compGOenrich, showCategory = 20, title = "Gene Ontology Enrichment Analysis")
rm(compGOenrich)
```

```{r}
compDO <- compareCluster(geneCluster = genes, fun= "enrichDO", pvalueCutoff = 0.05, pAdjustMethod = "BH")
```

```{r,fig.width=7.5,fig.height=15}
plot(compDO, showCategory = 20, title = "Disease ontology Enrichment Analysis")
rm(compDO)
```

```{r}
genes= lapply(peakAnnoList, function(i) as.data.frame(i)$geneId)
vennplot(genes)
```

Let's check how wide the peaks for each experiment are on average:
```{r}
test <- as.data.frame(ranges(peaks[[1]]))
print(names(peaks)[[1]])
mean(test$width)
test <- as.data.frame(ranges(peaks[[2]]))
print(names(peaks)[[2]])
mean(test$width)
test <- as.data.frame(ranges(peaks[[3]]))
print(names(peaks)[[3]])
mean(test$width)
test <- as.data.frame(ranges(peaks[[4]]))
print(names(peaks)[[4]])
mean(test$width)
test <- as.data.frame(ranges(peaks[[5]]))
print(names(peaks)[[5]])
mean(test$width)
```

Let's subset just the peaks from Nup98 and DBPs that interact within +/- 10kb of TSSs.

```{r}
Bound_promoters <- endoapply(peaks, subsetByOverlaps, query=promoter,ignore.strand=TRUE)
```

What proportion of the peaks is within the +/- 10kb region for each protein?
```{r}
prom_num=lapply(X = Bound_promoters, FUN = function(x) length(ranges(x)))
total_num=lapply(X = peaks, FUN = function(x) length(ranges(x)))
mapply("/",prom_num,total_num,SIMPLIFY = TRUE)
```

Now let's test if the overlap between Nup98 and DBPs is larger if we use only the peaks from promoter regions.


```{r}
biocLite("regioneR")
library(regioneR)
pt <- overlapPermTest(Bound_promoters[[1]], Bound_promoters[[5]], ntimes=1000, genome="hg19", count.once=TRUE)
pt
```

We need to save the GRanges as bed files to do that.
```{r}
setwd("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/ChIPseq/bed promoters")
rtracklayer::export.bed(Bound_promoters[[1]], "N98FLpromoters.bed")
rtracklayer::export.bed(Bound_promoters[[2]], "N98Ntpromoters.bed")
rtracklayer::export.bed(Bound_promoters[[3]], "DDX21promoters.bed")
rtracklayer::export.bed(Bound_promoters[[4]], "DDX5promoters.bed")
rtracklayer::export.bed(Bound_promoters[[5]], "DDX11promoters.bed")
```

We also need to create a TxDb file made only of the promoter regions of genes for random sampling.

```{r}
setwd("C:/Users/Juliana/Desktop/extra view/Nup98-DBPs_extraview/ChIPseq/bed promoters")
promoterCo <- getPromoters(TxDb=txdb, upstream=15000, downstream=15000)

promoterCO=c(resize(promoterCo, 10000, fix="center", use.names=TRUE, ignore.strand=FALSE),resize(promoterCo, 10000, fix="start", use.names=TRUE, ignore.strand=FALSE),resize(promoterCo, 10000, fix="end", use.names=TRUE, ignore.strand=FALSE))
```

```{r}
promoterCO2=c(resize(promoterCO, 3334, fix="center", use.names=TRUE, ignore.strand=FALSE),resize(promoterCO, 3334, fix="start", use.names=TRUE, ignore.strand=FALSE),resize(promoterCO, 3334, fix="end", use.names=TRUE, ignore.strand=FALSE))
```

```{r}
# Don't run for now...
promoterCO3=c(resize(promoterCO2, 1112, fix="center", use.names=TRUE, ignore.strand=FALSE),resize(promoterCO2, 1112, fix="start", use.names=TRUE, ignore.strand=FALSE),resize(promoterCO2, 1112, fix="end", use.names=TRUE, ignore.strand=FALSE))
```

```{r}
test1 <- asGFF(txdb)
test <- makeTxDbFromGRanges(getPromoters(TxDb=txdb, upstream=15000, downstream=15000), drop.stop.codons=FALSE, metadata=NULL, taxonomyId=9606)
```

```{r}
setwd("C:/Users/Juliana/Desktop/ChIPseq final/bed promoters")
files <- list.files()
files
```

Nucleoplasmic Nup98 DamID-seq:

```{r}
setwd("C:/Users/Juliana/Desktop/ChIPseq final/bed promoters")
enrichPeakOverlap(queryPeak = files[5], targetPeak = files[1:4], TxDb = promoterCO2, pAdjustMethod = "fdr", nShuffle = 1000, pool= TRUE, chainFile = NULL, verbose = FALSE)
```

Nucleoplasmic Nup98 DamID-seq:

```{r}
setwd("C:/Users/Juliana/Desktop/ChIPseq final/bed promoters")
enrichPeakOverlap(queryPeak = files[5], targetPeak = files[1:3], TxDb = txdb, pAdjustMethod = "fdr", nShuffle = 1000, pool= FALSE, chainFile = NULL, verbose = FALSE)
```

Full length Nup98 DamID-seq:

```{r}
enrichPeakOverlap(queryPeak = files[4], targetPeak = files[c(1:3,5)], TxDb = promoterCO2, pAdjustMethod = "BH", nShuffle = 1000, pool= TRUE, chainFile = NULL, verbose = FALSE)
```

Full length Nup98 DamID-seq:

```{r}
setwd("C:/Users/Juliana/Desktop/ChIPseq final/bed promoters")
enrichPeakOverlap(queryPeak = files[4], targetPeak = files[c(1:3,5)], TxDb = txdb, pAdjustMethod = "fdr", nShuffle = 1000, pool= FALSE, chainFile = NULL, verbose = FALSE)
```

Full length Nup98 DamID-seq:

```{r}
setwd("C:/Users/Juliana/Desktop/ChIPseq final/bed promoters")
enrichPeakOverlap(queryPeak = files[4], targetPeak = files[c(1:3,5)], TxDb = txdb, pAdjustMethod = "fdr", nShuffle = 1000, pool= TRUE, chainFile = NULL, verbose = FALSE)
```
Now